src/clj/harja/palvelin/integraatiot/tloik/ilmoitukset.clj

```diff
@@ -72,31 +74,67 @@
+  (let [tunnit (-> sekunnit (/ 3600) Math/floor int)
+        minuutit (-> sekunnit (- (* tunnit 3600)) (/ 60) Math/floor int)
+        sekunnit (-> sekunnit (- (* tunnit 3600) (* minuutit 60)) Math/floor int)]
+    (str tunnit "h " minuutit "min " sekunnit "s")))
+
+(defn- slack-viesti
+  [{:keys [kulunut-aika viesti-id tapahtuma-id kehitysmoodi?]}]
+  (let [url (if kehitysmoodi? "http://localhost:3000/"
+                              "https://extranet.liikennevirasto.fi/harja/")
+        integraatio-log-url (str url "#hallinta/integraatioloki?tapahtuma-id=" tapahtuma-id
+                                 "&alkanut=" (pvm/pvm->iso-8601 (pvm/nyt-suomessa))
+                                 "&valittu-jarjestelma=tloik&valittu-integraatio=ilmoituksen-kirjaus")]
+    (log/warn {:fields [{:title "Linkki"
+                         :value (str "<" integraatio-log-url "|Harja integraatio loki>")}]
+               :tekstikentta (str "Ilmoitukset ovat hitaita! :snail: :envelope:|||"
+                                  "Ilmoituksella, jonka viesti id on " viesti-id "|||"
+                                  "Kesti *" (ilmoituksen-kesto kulunut-aika) "* saapua T-LOIK:ista HARJAA:n")})))
+
 (defn kasittele-ilmoitus
   "Tallentaa ilmoituksen ja tekee tarvittavat huomautus- ja ilmoitustoimenpiteet"
   [sonja ilmoitusasetukset lokittaja db tapahtumat kuittausjono urakka
-   ilmoitus viesti-id korrelaatio-id tapahtuma-id jms-lahettaja]
-  (let [ilmoitus (assoc ilmoitus :urakkanimi (:nimi urakka))
-        urakka-id (:id urakka)
-        ilmoitus-id (:ilmoitus-id ilmoitus)
-        paivystajat (yhteyshenkilot/hae-urakan-tamanhetkiset-paivystajat db urakka-id)
-        kuittaus (kuittaus-sanoma/muodosta viesti-id ilmoitus-id (time/now) "valitetty" urakka
-                                           paivystajat nil)
-        ilmoittaja-urakan-urakoitsijan-organisaatiossa? (kayttajat-q/onko-kayttaja-nimella-urakan-organisaatiossa?
-                                                          db urakka-id ilmoitus)
-        uudelleen-lahetys? (ilmoitukset-q/ilmoitus-loytyy-viesti-idlla? db ilmoitus-id viesti-id)
-        ilmoitus-kanta-id (ilmoitus/tallenna-ilmoitus db urakka-id ilmoitus)
-        ilmoitus (assoc ilmoitus :id ilmoitus-kanta-id)
-        tieosoite (ilmoitus/hae-ilmoituksen-tieosoite db ilmoitus-kanta-id)]
-    (notifikaatiot/ilmoita-saapuneesta-ilmoituksesta tapahtumat urakka-id ilmoitus-id)
-    (if ilmoittaja-urakan-urakoitsijan-organisaatiossa?
-      (merkitse-automaattisesti-vastaanotetuksi db ilmoitus ilmoitus-kanta-id jms-lahettaja)
-      (when (not uudelleen-lahetys?)
-        (laheta-ilmoitus-paivystajille db
-                                      (assoc ilmoitus :sijainti (merge (:sijainti ilmoitus) tieosoite))
-                                      paivystajat urakka-id ilmoitusasetukset)))
+   ilmoitus viesti-id korrelaatio-id tapahtuma-id jms-lahettaja kehitysmoodi?
+   vastaanotettu]
+  (jdbc/with-db-transaction [db db]
```

Comment:
Ilmoituksen tallennustransaktioon on nyt lisätty jonkin verran koodia, poikkeukset tässä lohkossa peruvat ilmoitustransaktion. Pitäisikö uutta koodia muotoilla niin että
käsittelynopeus lasketaan vasta kun on tallennettu ilmoitus, lähetetään se eteenpäin ja suljetaan transaktio.